ui/hooks/useTokensToSearch.js

```diff
@@ -71,9 +71,12 @@ export function getRenderableTokenData(
     // And since the token.address from allTokens is checksumaddress
     // token Address have to be changed to lowercase when we are using dynamic list
     tokenAddress = useTokenDetection ? address?.toLowerCase() : address;
-    tokenIconUrl = useTokenDetection
-      ? tokenList[tokenAddress]?.iconUrl
-      : `images/contract/${tokenList[tokenAddress].iconUrl}`;
+
+    if (useTokenDetection) {
+      tokenIconUrl =
+        tokenList[tokenAddress]?.iconUrl ||
+        `images/contract/${tokenList[tokenAddress]?.iconUrl}`;
+    }
```

Comment:
in the previous code, if useTokenDetection was false, then 
```
tokenIconUrl = `images/contract/${tokenList[tokenAddress]?.iconUrl}`
```

should we preserve this behavior?

Reply:
I'm wondering if it's supposed to be `undefined` when `iconUrl` is not present though. Because I imagine `images/contract/undefined` is not a valid icon?

Reply:
Perhaps
```
tokenIconUrl = tokenList[tokenAddress]?.iconUrl
if(!useTokenDetection){
    tokenIconUrl = `images/contract/${tokenIconUrl}`
}
```

Reply:
Oh yes you're right @digiwand, I mistakenly changed the behavior. Correction that right now

Reply:
Yes that's a good catch @FrederikBolding

Reply:
This may be a problem elsewhere too from looking at the commit

Reply:
Oh, good catch @FrederikBolding